pkg/segment/query/processor/queryprocessor.go

```diff
@@ -110,7 +110,47 @@ func NewQueryProcessor(firstAgg *structs.QueryAggregators, queryInfo *query.Quer
 		dataProcessors[i].streams = append(dataProcessors[i].streams, NewCachedStream(dataProcessors[i-1]))
 	}
 
-	var lastStreamer streamer = searcher
+	/*
+		1. If the query is a bottleneck and is Distributed, then the query will be distributed until the bottleneck.
+		2. Once bottleneck index is identified, split the data processors into two parts:
+			- The first part which contains the data processors until the bottleneck, will be distributed.
+				-  Call dqs.GetStreams by passing the first part of the data processors.
+				-  Hook up the streams to the first data processor in the second part.
+				-  this will make : ([dataProcessorWorker1.streams(dataProcessorFirstPart), dataProcessorWorker2.streams(dataProcessorFirstPart), ...],
+					dataProcessorSecondPart[0], ... dataProcessorSecondPart[n-1])
+			- The second part which contains the data processors after the bottleneck, will be executed locally.
+	*/
+	if queryInfo.IsDistributed() {
+		splitIndex, shouldSplit := getSplitIndex(dataProcessors, queryType)
+		var chainedDp1, chainedDp2 []*DataProcessor
+		if shouldSplit {
+			chainedDp1 = dataProcessors[:splitIndex+1] // +1 to include the splitIndex data processor.
+			chainedDp2 = dataProcessors[splitIndex+1:] // +1 to exclude the splitIndex data processor.
+		} else {
+			chainedDp1 = dataProcessors
+		}
+
+		streamsAsAny := queryInfo.GetDQS().GetDistributedStreams(chainedDp1, searcher, queryInfo)
+		if streamsAsAny != nil {
+			streams, ok := streamsAsAny.([]*CachedStream)
+			if ok {
+				if len(chainedDp2) == 0 {
+					// Since there are no data processors after the bottleneck,
+					// we will hook up an EOF data processor as the bottleneck command
+					// fetches all the data from all the worked node streams.
+					chainedDp2 = []*DataProcessor{NewEofDP()}
+				}
```

Comment:
I'm a little confused. Do you need a pass through DP here instead of an EOF DP? I'm not sure of the purpose of the EOF DP; it seems like it would give incorrect results since it always returns EOF but there may be more data to fetch??

Reply:
Yes I changed it to `passThroughDP`